presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -303,6 +306,11 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
                     Optional.ofNullable(sessionContext.getSchema()),
                     sessionContext.getIdentity().getPrincipal().map(Principal::getName)));
 
+            session = sessionBuilder.build();
+            if (sessionContext.getTransactionId().isPresent()) {
```

Comment:
I reviewed the tasks that happen within beginTransactionId and I don't believe there is any dependency between what happens in there and the work done to create preparedQuery. Implying that the order does not matter on if we prepare the query or if we begin the transaction. From testing, I also found no impact in performance or functionality.

Reply:
why are we starting a transaction here ?

Reply:
The transaction was never started after I refactored the code to use SessionBuilder instead of Session. In the original construction of Session, the transaction was started once Session was built. Since we need queryType in the session, I moved out the transaction piece out so we could utilize SessionBuilder to add the necessary field. Since the transaction hasn't started, I moved it out and placed it as early as I could while including the addition of queryType.

Reply:
i think the transaction has to begin before you call analyzer ..so this is too late